iphone/Maps/Core/Location/MWMLocationManager.mm

```diff
@@ -469,43 +484,24 @@ - (void)setStarted:(BOOL)started
   }
 }
 
-- (BOOL)start
+- (void)requestPermissionIfNeededFor:(CLLocationManager *)manager
 {
-  MWMVoidBlock doStart = ^{
-    LOG(LINFO, ("startUpdatingLocation"));
-
-    CLLocationManager * locationManager = self.locationManager;
-    if ([locationManager respondsToSelector:@selector(requestWhenInUseAuthorization)])
-      [locationManager requestWhenInUseAuthorization];
-
-    [locationManager startUpdatingLocation];
-
-    setPermissionRequested();
+  [manager requestWhenInUseAuthorization];
```

Comment:
If there is only one line in this method, would it be better to call this line directly, without introducing a wrapping function?

Reply:
Good point!